src/components/collections/Listing.js

```diff
@@ -0,0 +1,129 @@
+import { queryCache, useQuery } from "react-query";
+
+import BasicTable from "components/table/BasicTable";
+import WrappedErrorHandler from "components/error/WrappedErrorHandler";
+import isQueryLoading from "components/utils/isQueryLoading";
+import { useUserProfile } from "contexts/userProfile";
+import { useTranslation } from "i18n";
+import ids from "./ids";
+import { COLLECTION_FILTER } from "./index";
+import {
+    ALL_COLLECTIONS_QUERY,
+    getAllCollections,
+    getMyCollections,
+    MY_COLLECTIONS_QUERY,
+} from "serviceFacades/groups";
+import DELink from "../utils/DELink";
+
+function Columns(t) {
+    return {
+        NAME: { fieldName: t("name"), key: "display_extension" },
+        DESCRIPTION: { fieldName: t("description"), key: "description" },
+    };
+}
+
+function Listing(props) {
+    const { parentId, filter, onCollectionSelected } = props;
+
+    const { t } = useTranslation(["collections", "search"]);
+    const [data, setData] = useState(null);
+    const [userProfile] = useUserProfile();
+
+    const tableId = buildID(parentId, ids.TABLE);
+    const COLLECTION_COLUMNS = Columns(t);
+
+    const myCollectionsCache = queryCache.getQueryData([
+        MY_COLLECTIONS_QUERY,
+        { userId: userProfile?.id },
+    ]);
+
+    useEffect(() => {
+        if (
+            myCollectionsCache &&
+            (!data || COLLECTION_FILTER.MY_COLLECTIONS === filter)
+        ) {
+            setData(myCollectionsCache.groups);
+        }
+    }, [data, filter, myCollectionsCache]);
+
+    const columns = useMemo(() => {
+        return [
+            {
+                Header: COLLECTION_COLUMNS.NAME.fieldName,
+                accessor: COLLECTION_COLUMNS.NAME.key,
+                Cell: ({ row, value }) => {
+                    const collection = row.original;
+                    const rowId = buildID(tableId, collection.id);
+                    return (
+                        <DELink
+                            id={buildID(rowId, ids.COLLECTION_LINK)}
+                            onClick={() => onCollectionSelected(collection)}
+                            text={value}
+                        />
+                    );
+                },
+            },
+            {
+                Header: COLLECTION_COLUMNS.DESCRIPTION.fieldName,
+                accessor: COLLECTION_COLUMNS.DESCRIPTION.key,
+            },
+        ];
+    }, [COLLECTION_COLUMNS, onCollectionSelected, tableId]);
+
+    const { isFetching: fetchMyCollections, error: myCollectionsError } =
+        useQuery({
+            queryKey: [MY_COLLECTIONS_QUERY, { userId: userProfile?.id }],
+            queryFn: getMyCollections,
+            config: {
+                enabled:
+                    !myCollectionsCache &&
+                    COLLECTION_FILTER.MY_COLLECTIONS === filter,
+                onSuccess: (results) => {
+                    setData(results.groups);
+                },
+            },
+        });
```

Comment:
For caching, usually I follow these steps:
 - Increase cache / stale time for the related query (or set it to infinity)
 - Try to get the query data from cache when needed. if thats not available, then fetch it 
 - Update query data when new data is available or simply invalidate that query. 

Not sure if am missing any thing but I didn't see the step 1 happening here. Also, Not sure how things are will RQ3 as well 🤔 since queryCache is gone and I saw some weirdness with bags 😖 in RQ3 .

Reply:
For step 1, I think we're increasing the cache/stale time to Infinity in cases where we only want to fetch the query once, and if it's successful, never fetch it again (e.g. bootstrap, tool types). 
In this case, I want it to stay cached across the 2 pages (collections and apps) until the user (un)follows, creates, or deletes a new collection.  I might have missed something, but when I tried locally adding the cache/stale time and setting them to Infinity, and invalidated the query during those mutations, the query wouldn't refresh. I'd still get back the old list that was previously cached.
If this ends up being an issue with RQ3, we can either remove the caching (I just thought it was more convenient to have it cached so the user doesn't have to keep waiting for the list to populate) or we can switch to using a context.

Reply:
Ok. We can revisit again after the RQ3 update.